app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java

```diff
@@ -0,0 +1,199 @@
+package com.kabouzeid.gramophone.views;
+
+import android.content.Context;
+import android.graphics.Rect;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.widget.FrameLayout;
+import android.widget.HorizontalScrollView;
+import android.widget.TextView;
+
+import com.kabouzeid.gramophone.R;
+
+/**
+ * @author Lincoln (theduffmaster)
+ * 
+ * A custom FrameLayout view that intercepts touch events and decides whether to consume them or
+ * pass on the touch events to its children.
+ */
+public class TouchInterceptFrameLayout extends FrameLayout {
+
+    private static final int MAX_CLICK_DISTANCE = 5;
+    private static final int MAX_VERTICAL_DISTANCE = 5;
+
+    private Context c;
+
+    private String TAG;
+    private String NULL_VIEWS_EXCEPTION_MESSAGE;
+    
+    private HorizontalScrollView scrollView;
+    private TextView textView;
+    private Rect scrollViewRect = new Rect();
+    private float startX;
+    private float startY;
+    private boolean isTap;
+    private String songTruncated;
+    private String song;
+
+    public TouchInterceptFrameLayout(@NonNull Context context) {
+        super(context);
+        c= context;
+        TAG = c.getResources().getString(R.string.TOUCH_INTERCEPT_FRAME_LAYOUT_TAG);
+        NULL_VIEWS_EXCEPTION_MESSAGE = c.getResources().getString(R.string.NULL_VIEWS_EXCEPTION_MESSAGE);
+    }
+
+    public TouchInterceptFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+        c= context;
+        TAG = c.getResources().getString(R.string.TOUCH_INTERCEPT_FRAME_LAYOUT_TAG);
+        NULL_VIEWS_EXCEPTION_MESSAGE = c.getResources().getString(R.string.NULL_VIEWS_EXCEPTION_MESSAGE);
+    }
+
+    public TouchInterceptFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        c= context;
+        TAG = c.getResources().getString(R.string.TOUCH_INTERCEPT_FRAME_LAYOUT_TAG);
+        NULL_VIEWS_EXCEPTION_MESSAGE = c.getResources().getString(R.string.NULL_VIEWS_EXCEPTION_MESSAGE);
+    }
+
+    /**
+     * Fetches the HorizontalScrollView contained by this FrameLayout and also gets the
+     * TextView that is contained within that HorizontalScrollView.
+     * Must be called before setTruncateText.
```

Comment:
Why not just merge the two then? You could pass in the text to this method as well.